### DISCUSSION TITLE
Make it possible to pass a path to a file with the connection string

### DISCUSSION THREAD

**Discussion by eden-lane** [prisma/prisma#4979] [Q&A] [answered]
In your prisma file you have to specify an environment variable where you store connection url. Like this:
```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}
```
I'm using [docker secrets](https://docs.docker.com/engine/swarm/secrets/) to store my sensitive data (including connection url). When you use docker secrets it mounts a file with a secret into your container. Basically you can open a file `/run/secrets/database_url` and you will see nothing but a connection url.

So my suggestion is to add "file" command to prisma configuration file so you can read a connection string from file. Something like this:
```prisma
datasource db {
  provider = "postgresql"
  url      = file("/run/secrets/database_url")
}
``` 

**Why?**:
There is no other _secure_ way to store env variable in your docker stack. Having .env file in your container is [dangerous](https://codeburst.io/how-secure-is-your-environment-file-in-node-js-7c4d2ed0d15a).

**Comment by nikolasburk** [comment, marked as answer]
Hey @eden-lane, thanks a lot for opening this discussion! This looks like a great idea, would you mind reposting it as a proper [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) so that our Product and Engineering teams can pick it up?

**Reply by eden-lane** [reply]
Thanks! I will do 👌

**Reply by eden-lane** [reply]
done:
#4980

**Reply by nikolasburk** [reply]
Thank you so much! 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
